mozilla-services/socorro
# bug-1889185: implement stage submitter and fake collector

This implements a stage submitter that pulls crash ids from QUEUE and crash data from STORAGE, packages that up into a crash report payload, and submits it to STAGE_SUBMITTER_DESTINATIONS.

This creates a `make runsubmitter` system that runs the stage submitter and a fake collector allowing you to use `bin/process_crash.sh` to set everything up and publish a crash id to the standard queue which the submitter will see and submit a crash to the fake collector. The fake collector will parse the payload and log headers, annotations, and files.

This adds stage submitter tests.

To test:

1. `make build`
2. `make setup`
3. `make runservices`
4. `make runsubmitter`
5. in another terminal, run `make shell` and then you can use `bin/process_crash.sh` to download crash data, put it in storage, and publish the crash id to the standard queue
   1. if it's sampled, then you'll see a `socorro.submitter.sample` incr metric
   2. if it's accepted, then you'll see a `socorro.submitter.accept` incr metric and you'll see the headers, annotations, and files logged by the fake collector

Light documentation is provided in `docs/services/stage_submitter.rst`.
